### PR TITLE
tabs, tab, tabpanel unit tests

### PR DESCRIPTION
**Pull request by brianchristopherbrady**
Adds tabs, tab, and tabpanel unit tests

### CODE REVIEW

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a2759700ee41843a4534edbfbf00975a21226bf8'>a2759700ee41843a4534edbfbf00975a21226bf8</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/web-components/src/tabs/tabs.spec.ts [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+
+      await tab.evaluate((node, i) => {
+        node.id = `custom-id-${i}`;
+      }, i);
+
+      await expect(tab).toHaveAttribute('id', `custom-id-${i}`);
+    }
+  });
+
+  test('should set an `id` attribute on tab items with a unique ID when an `id` is NOT provided', async () => {
```
Comment (halerankin) [inline packages/web-components/src/tabs/tabs.spec.ts:151]: Maybe group all the "when ID is NOT provided" into it's own test.describe( ... ), then test each scenario within.

File: packages/web-components/src/tabs/tabs.spec.ts [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+
+      await expect(tabPanel).toHaveCount(1);
+
+      const tabId = (await tab.getAttribute('id')) as string;
+
+      expect(tabId).toMatch(/tab-\d+/);
+
+      await expect(tabPanel).toHaveAttribute('aria-labelledby', tabId);
+
+      await expect(tab).toHaveAttribute('aria-controls', panelId);
```
Comment (halerankin) [inline packages/web-components/src/tabs/tabs.spec.ts:233]: Dang, this one looks crazy complex! Way to go!!

**Review by halerankin — APPROVED** [review]
w/ comment

File: packages/web-components/src/tabs/tabs.spec.ts [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+                  <fluent-tabs></fluent-tabs>
+              `;
+    });
+
+    await expect(element).toHaveJSProperty('orientation', 'horizontal');
+  });
+
+  test('should set an `id` attribute on the active tab when an `id` is provided', async () => {
```
Comment (miroslavstastny) [inline packages/web-components/src/tabs/tabs.spec.ts:128]: nit: What do you mean by the "active" tab? the test set and tests the `id` on all tabs.

**Review by miroslavstastny — COMMENTED** [review]
The tests lgtm.
But lint is not happy yet:
```
packages/web-components/src/tabs/tabs.spec.ts
    5:1   error    `../tab/tab.js` import should occur before import of `./tabs.js`  import/order
  294:27  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  315:27  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  336:27  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  437:29  warning  Unexpected use of { force: true } option                          playwright/no-force-option
  462:24  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  464:25  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  472:29  warning  Unexpected use of { force: true } option                          playwright/no-force-option
  482:29  warning  Unexpected use of { force: true } option                          playwright/no-force-option
  507:24  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  509:24  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  540:24  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
  542:25  warning  Avoid having conditionals in tests                                playwright/no-conditional-in-test
```

**Review by miroslavstastny — APPROVED** [review]

**Pull request merged**